internal/db/reset/reset.go

```diff
@@ -217,7 +220,22 @@ func restartServices(ctx context.Context) error {
 }
 
 func listServicesToRestart() []string {
-	return []string{utils.StorageId, utils.GotrueId, utils.RealtimeId, utils.PoolerId}
+	var services []string
+
+	if utils.Config.Storage.Enabled {
+		services = append(services, utils.StorageId)
+	}
+	if utils.Config.Realtime.Enabled {
+		services = append(services, utils.RealtimeId)
+	}
+	if utils.Config.Db.Pooler.Enabled {
+		services = append(services, utils.PoolerId)
+	}
+	if utils.Config.Auth.Enabled {
+		services = append(services, utils.GotrueId)
+	}
+
+	return services
```

Comment:
I think this is unnecessary because if a service isn't already started, restarting it will throw 404 which is then ignored by cli https://github.com/supabase/cli/pull/2663/files#diff-e7e370d31621951333d60503924eb420ab55bf0bc8e903e7b86e1766baef60edR213

Additionally if user runs supabase start, then disable a service in config.toml, running db reset should probably stop the service instead of restarting. But to avoid complicating the logic, I think it's ok to let users manually stop and restart.